### PR TITLE
Fix Wx inconsistencies

### PR DESCRIPTION
**Pull request by DietmarSchwertberger**
## PR Summary
From discussion at PR 10428:
The wx examples in master are currently not consistent with the recent changes around NavigationToolbar2.
This PR makes NavigationToolbar2 consistent through the three wx backends.
Examples are adjusted to be similar to the documentation example at https://matplotlib.org/users/navigation_toolbar.html
The unused and undocumented class `SubplotToolWX` has been removed (Google did not find any code using this.)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_wx.py [hunk 1]
```diff
@@ -1474,25 +1474,6 @@ def updateButtonText(self, lst):
 }
 
 
-class SubplotToolWX(wx.Frame):
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_wx.py]: API changes (even for "apparently unused" stuff) should nearly always go through a deprecation period unless it's really too hard to keep around.  Just decorate the class with `@deprecated("2.2")`.

File: lib/matplotlib/backends/backend_wxagg.py [hunk 2]
```diff
@@ -71,15 +71,8 @@ def blit(self, bbox=None):
     filetypes = FigureCanvasAgg.filetypes
 
 
-@cbook.deprecated("2.2")
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_wxagg.py]: Please leave until 3.0 (one minor-release deprecation period at least, usually two).

File: lib/matplotlib/backends/backend_wxcairo.py [hunk 3]
```diff
@@ -6,8 +6,8 @@
 import wx
 
 from .backend_cairo import cairo, FigureCanvasCairo, RendererCairo
-from .backend_wx import (
-    _BackendWx, _FigureCanvasWxBase, FigureFrameWx, NavigationToolbar2Wx)
+from .backend_wx import _BackendWx, _FigureCanvasWxBase, FigureFrameWx
+from .backend_wx import NavigationToolbar2Wx as NavigationToolbar2WxCairo
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_wxcairo.py]: can go in the previous line (would prefer this) as
```
from .backend_wx import (
    _BackendWx, _FigureCanvasWxBase, FigureFrameWx,
    NavigationToolbar2Wx as NavigationToolbar2WxCairo)
```

File: lib/matplotlib/backends/backend_wxagg.py (hunk 2 above)
Comment (DietmarSchwertberger) [reply lib/matplotlib/backends/backend_wxagg.py]: I don't see that any other NavigationToolbar2 is deprecated, especially not the wx non-Agg one.
With the derived class removed and the modified `import NavigationToolbar2Wx as NavigationToolbar2WxAgg` things are consistent again (no deprecation).

Not having the Toolbar alias in backend_wxagg can break things, though.

File: lib/matplotlib/backends/backend_wxcairo.py (hunk 3 above)
Comment (DietmarSchwertberger) [reply lib/matplotlib/backends/backend_wxcairo.py]: Done, also for wxagg.

File: lib/matplotlib/backends/backend_wx.py (hunk 1 above)
Comment (DietmarSchwertberger) [reply lib/matplotlib/backends/backend_wx.py]: Done

File: lib/matplotlib/backends/backend_wxagg.py (hunk 2 above)
Comment (DietmarSchwertberger) [reply lib/matplotlib/backends/backend_wxagg.py]: Having a deprecation for the Toolbar alias would require to replace the assignment `Toolbar = NavigationToolbar2Wx` with a derived and deprecated class. I would now be in favor of having this in wx and wxagg. What do you think?

Comment (anntzer) [reply lib/matplotlib/backends/backend_wxagg.py]: derived deprecated class sounds good to me.

Comment (DietmarSchwertberger) [reply lib/matplotlib/backends/backend_wxagg.py]: Thanks. Done, incl. alternatives in deprecation messages.

**Comment by DietmarSchwertberger** [comment]
"Release critical" together with PR #10428, to have working wx backends.

**Pull request merged**